CMakeLists.txt

```diff
@@ -262,6 +269,21 @@ add_library(bitsandbytes SHARED ${SRC_FILES})
 target_compile_features(bitsandbytes PUBLIC cxx_std_17)
 target_include_directories(bitsandbytes PUBLIC csrc include)
 
+if (BUILD_CPU)
+    target_link_libraries(bitsandbytes PRIVATE OpenMP::OpenMP_CXX)
+    include(CheckCXXCompilerFlag)
+
+    check_cxx_compiler_flag(-mavx512f HAS_AVX512F)
+    check_cxx_compiler_flag(-mavx512bf16 HAS_AVX512BF16)
+
+    if(HAS_AVX512F)
+    target_compile_options(bitsandbytes PRIVATE -mavx512f)
+    endif()
+
+    if(HAS_AVX512BF16)
+    target_compile_options(bitsandbytes PRIVATE -mavx512bf16)
+    endif()
+endif()
```

Comment:
Hi @jiqing-feng we still have some build issues with this.

A few things need consideration here:
* We build for Linux aarch64
* We also build for macOS arm64. I'm not sure how to use OpenMP on that platform - maybe we can skip for now?
* On Windows x86-64 we build with MSVC. Apart from `/arch:AVX512` I don't really know if there is a flag for AVX512-BF16.

Reply:
I have updated the cmake file, please check it. Thanks.